### PR TITLE
feat: rediscluster support hostnetwork and custom port

### PR DESCRIPTION
**Pull request by drivebyer**
**Description**

Support hostnetwork for redis cluster mode.

> ps: only support with clusterVersion: v6. Because we can not connect to hostnetwork pod with FQDN name

#183 

**Type of change**

* New feature (non-breaking change which adds functionality)

**Checklist**

- [x] Tests have been added/modified and all tests pass.
- [x] Functionality/bugs have been confirmed to be unchanged or fixed.
- [x] I have performed a self-review of my own code.
- [ ] Documentation has been updated or added where necessary.

**Additional Context**

Here are test case:

1. create cluster:
截屏2023-12-12 18 14 57
2. check cluster:
截屏2023-12-12 18 15 33
3. cluster cluster out of k8s:
截屏2023-12-12 18 17 45

### CODE REVIEW

**Comment by shubham-cmyk** [comment]
@drivebyer Does this solves the issue for the connecting outside ?

File: api/v1beta2/rediscluster_types.go [hunk 1]
```diff
@@ -27,6 +27,9 @@ import (
 type RedisClusterSpec struct {
 	Size             *int32           `json:"clusterSize"`
 	KubernetesConfig KubernetesConfig `json:"kubernetesConfig"`
+	HostNetwork      bool             `json:"hostNetwork,omitempty"`
+	// +kubebuilder:default:=6379
+	Port int32 `json:"port,omitempty"`
```
Comment (shubham-cmyk) [inline api/v1beta2/rediscluster_types.go:32]:
```suggestion
	Port  *int `json:"port,omiteimpty"`
```

Comment (shubham-cmyk) [reply api/v1beta2/rediscluster_types.go:32]: Make this pointer we can check if the port==nil then we se the default instead of checking it of 
port==0

**Comment by drivebyer** [comment]
> @drivebyer Does this solves the issue for the connecting outside ?

Sure, see `Additional Context` of current PR.

**Comment by shubham-cmyk** [comment]
> > @drivebyer Does this solves the issue for the connecting outside ?
> 
> Sure, see `Additional Context` of current PR.

Looks Good.

**Comment by drivebyer** [comment]
@shubham-cmyk  ready now

**Review by shubham-cmyk — APPROVED** [review]
LGTM

**Pull request merged**